modules/reference/pages/data-transform-rust-sdk.adoc

```diff
@@ -0,0 +1,130 @@
+= Rust SDK for Data Transforms
+:description: Work with data transforms using Rust.
+
+Rust SDK reference documentation for Redpanda data transforms.
+
+== Data transforms
+
+The `redpanda_transform_sdk` crate contains the core functions and types for transforming data within Redpanda.
+
+=== Data transforms functions
+
+==== on_record_written
+
+[,rust,role="no-copy"]
+----
+pub fn on_record_written<E, F>(cb: F) -> !
+where
+    E: Debug,
+    F: Fn(WriteEvent<'_>, &mut dyn RecordWriter) -> Result<(), E>,
+----
+
+The `on_record_written` function transforms individual records after they have been written to an input topic. The function registers a callback to be fired when a record is written to the topic. This callback is triggered after the record has been written and fsynced to disk and the producer has been acknowledged.
+
+Since this method blocks and runs forever, it should be called from the `main` function. Perform the necessary steps prior to transforming any records before calling this method.
```

Comment:
```suggestion
Since this method blocks and runs forever, it should be called from the `main()` function. Perform the necessary steps before transforming any records before calling this method.
```

Reply:
Google style guide suggests using `()` after function/method names to make it clearer that they are executable: https://developers.google.com/style/code-in-text#method-names